internal/credential/vault/repository_credential_library_test.go

```diff
@@ -897,6 +1162,24 @@ func TestRepository_UpdateCredentialLibrary(t *testing.T) {
 			if tt.wantCount > 0 {
 				assert.NoError(db.TestVerifyOplog(t, rw, got.GetPublicId(), db.WithOperation(oplog.OpType_OP_TYPE_UPDATE), db.WithCreateNotBefore(10*time.Second)))
 			}
+
+			if tt.orig.MappingOverride != nil {
```

Comment:
This check invalidates the `no-mapping-override-change-username-and-password-attributes` test because it has a `orig.MappingOverride` of `nil`, but it does expect to have a mapping override after the update that should be checked.

Verifying this locally revealed an issue with this test returning early and never testing any of the override mappings, on line 1154 and line 1159 we return after doing a `dbassert.IsNull`, these should probably be updated to 
```
if tt.want.Name == "" {
	dbassert.IsNull(got, "name")
} else {
	assert.Equal(tt.want.Name, got.Name)
}
```

Reply:
Great catch! Thanks for reviewing the tests so thoroughly! I made changes to address the issues you found. Can you take another look?